crates/bevy_reflect/src/type_name.rs

```diff
@@ -0,0 +1,66 @@
+use crate::utility::GenericTypeNameCell;
+
+/// Provide the name of the type as string.
+pub trait TypeName {
+    fn name() -> &'static str;
```

Comment:
Could you add some documentation here as well? I would highlight the fact that this can be used in place of `std::any::type_name` to provide a "stable type name" (and probably define what that means lol).